lib/src/error.rs

```diff
@@ -19,3 +19,9 @@ pub enum RuntimeError {
     #[fail(display = "Stack {} is empty", 0)]
     EmptyStack(StackIdentifier),
 }
+
+#[derive(Debug, Fail)]
+pub enum ParseError {
+    #[fail(display = "Parse error: {}", 0)]
+    Generic(String),
+}
```

Comment:
Just a pro tip, whoever merges their PR second is gonna have to change this, so merge this with the `CompileError` that I made. We'll also have to change `Compiler::parse` to return a `Result<..., CompileError>` instead of a `Fallible`